gcp/api/server.py

```diff
@@ -739,11 +767,17 @@ def to_response(b):
       bugs[i].related[:] = sorted(
           list(set(related_bug_ids + list(bugs[i].related))))
 
-  if next_page_token:
-    next_page_token = next_page_token.urlsafe()
+  if context.query_counter < context.input_cursor.query_number:
+    # If the input cursor is for a query number that's greater than
+    # the number of queries performed, the cursor must be invalid
+    # (and there will be no results, as everything is skipped)
+    raise ValueError("Cursor is invalid/does not belong to this query")
```

Comment:
Do you want to include the `query_counter` and `input_cursor.query_number` values here to aid in any future debugging? (Or a log entry with them prior to raising the exception?)

Reply:
Good idea, added an error log.